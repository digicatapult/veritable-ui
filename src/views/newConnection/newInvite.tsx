import Html from '@kitajs/html'
import { singleton } from 'tsyringe'
import { COMPANY_NUMBER, companyNumberRegex, CountryCode, EMAIL } from '../../models/stringTypes.js'
import { Page } from '../common.js'
import { FormFeedback, NewConnectionTemplates } from './base.js'

export type NewInviteFormStage = 'form' | 'confirmation' | 'success'

@singleton()
export class NewInviteTemplates extends NewConnectionTemplates {
  constructor() {
    super()
  }

  public newInviteFormPage = (feedback: FormFeedback, registryCountryCode?: CountryCode) => {
    return (
      <Page
        title="Veritable - New Connection"
        heading="New Connection"
        activePage="connections"
        headerLinks={[
          { name: 'Connections', url: '/connection' },
          { name: 'Invite New Connection', url: '/connection/new' },
        ]}
        stylesheets={['connection.css']}
      >
        <div class="connections header">
          <span>Invite New Connection</span>
        </div>
        <div class="card-body">
          <this.newInviteForm feedback={feedback} formStage="form" registryCountryCode={registryCountryCode} />
        </div>
      </Page>
    )
  }

  public newInviteForm = (props: {
    formStage: NewInviteFormStage
    companyNumber?: COMPANY_NUMBER
    email?: EMAIL
    registryCountryCode?: CountryCode
    feedback: FormFeedback
  }): JSX.Element => {
    switch (props.formStage) {
      case 'form':
        return <this.newInviteInput {...props}></this.newInviteInput>
      case 'confirmation':
        return <this.newInviteConfirmation {...props}></this.newInviteConfirmation>
      case 'success':
        return <this.newInviteSuccess {...props}></this.newInviteSuccess>
    }
  }

  private newInviteInput = (props: {
    companyNumber?: COMPANY_NUMBER
    email?: EMAIL
    registryCountryCode?: CountryCode
    feedback: FormFeedback
  }): JSX.Element => {
    return (
      <this.newConnectionForm
        submitRoute="new/create-invitation"
        feedback={props.feedback}
        progressStep={1}
        progressStepCount={3}
        actions={[
          { type: 'link', text: 'Cancel', href: '/connection' },
          { type: 'submit', value: 'continue', text: 'Continue' },
        ]}
      >
        <div id="new-invite-country-with-code">
          <select
            id="new-invite-country-select"
            name="registryCountryCode"
            required
            value={props.registryCountryCode}
            hx-get="/connection/new"
            hx-trigger="change"
            hx-target="#dynamic-invite-form-content"
            hx-select="#dynamic-invite-form-content"
            hx-swap="outerHTML"
            hx-include="this"
            onchange="document.getElementById('new-invite-country-code-display').value = this.value;"
          >
            <option value="GB">United Kingdom</option>
            <option value="US">United States</option>
          </select>
          <input id="new-invite-country-code-display" type="text" readonly value={props.registryCountryCode} />
        </div>

        <div id="dynamic-invite-form-content">
          <p>Configured registries for this country </p>
          {props.feedback.type == 'message' && Html.escapeHtml(props.feedback.registryOptionsPerCountry) && (
            <div class="registry-selection">
              {/* Country Registry Button - only show if there are country registries */}
              {props.feedback.registryOptionsPerCountry.countryRegistries.length > 0 && (
                <div class="registry-option">
                  <input
                    id="country-registry-radio"
                    type="radio"
                    name="selectedRegistry"
                    value={props.feedback.registryOptionsPerCountry.countryRegistries[0].registry_key}
                    checked={true}
                  />
                  <label for="country-registry-radio">
                    {Html.escapeHtml(props.feedback.registryOptionsPerCountry.countryRegistries[0].registry_name)}
                  </label>
                </div>
              )}

              {/* Third Party Registry Button - only show if there are third party registries */}
              {props.feedback.registryOptionsPerCountry.thirdPartyRegistries.length > 0 && (
                <div class="registry-option">
                  <input
                    id="third-party-registry-radio"
                    type="radio"
                    name="selectedRegistry"
                    value={props.feedback.registryOptionsPerCountry.thirdPartyRegistries[0].registry_key}
                  />
                  <label for="third-party-registry-radio">
                    {Html.escapeHtml(props.feedback.registryOptionsPerCountry.thirdPartyRegistries[0].registry_name)}
                  </label>
                  <span
                    id="third-party-registry-info-icon"
                    title="Third-party registries are external data providers that aggregate company information from multiple sources. We use OpenCorporates, a comprehensive database of company information from around the world."
                  >
                    i
                  </span>
                </div>
              )}
            </div>
          )}

          <input
            id="new-invite-company-number-input"
            name="companyNumber"
            placeholder="Companie's Number or ID"
            required
            hx-get="/connection/new/verify-company"
            hx-trigger="keyup changed delay:200ms, change, load"
            hx-target="#new-connection-feedback"
            hx-select="#new-connection-feedback"
            hx-swap="outerHTML"
            hx-include="#dynamic-invite-form-content, #new-invite-country-select"
            pattern={
              props.feedback.type === 'message' && props.feedback.regex
                ? props.feedback.regex
                : companyNumberRegex.source
            }
            minlength={props.feedback.type === 'message' && props.feedback.minlength ? props.feedback.minlength : 6}
            maxlength={props.feedback.type === 'message' && props.feedback.maxlength ? props.feedback.maxlength : 10}
            oninput="this.reportValidity()"
            value={props.companyNumber}
            type="text"
          ></input>
          <input
            required
            id="new-invite-email-input"
            placeholder="Connection's Email Address"
            name="email"
            value={props.email}
            type="email"
          ></input>
        </div>
      </this.newConnectionForm>
    )
  }

  private newInviteConfirmation = (props: {
    companyNumber?: COMPANY_NUMBER
    email?: EMAIL
    feedback: FormFeedback
  }): JSX.Element => {
    return (
      <this.newConnectionForm
        submitRoute="new/create-invitation"
        feedback={props.feedback}
        progressStep={2}
        progressStepCount={3}
        actions={[
          { type: 'submit', value: 'back', text: 'Back' },
          { type: 'submit', value: 'submit', text: 'Submit' },
        ]}
      >
        <input
          id="new-invite-company-number-input"
          name="companyNumber"
          value={props.companyNumber}
          type="hidden"
        ></input>
        <input id="new-invite-email-input" name="email" value={props.email} type="hidden"></input>,
        {/* todo: add this onto SharedOrgInfo */}
        <input
          id="new-invite-selected-registry-input"
          name="selectedRegistry"
          value={props.feedback.type === 'companyFound' ? props.feedback.company.selectedRegistry : ''}
          type="hidden"
        />
        <input
          id="new-invite-registry-country-code-input"
          name="registryCountryCode"
          value={props.feedback.type === 'companyFound' ? props.feedback.company.registryCountryCode : ''}
          type="hidden"
        />
        <div id="new-connection-confirmation-text">
          <p>Please confirm the details of the connection before sending</p>
          <p>
            {' '}
            {Html.escapeHtml(props.feedback.type === 'companyFound' && props.feedback.company.registryCountryCode)}
          </p>
          <p>
            {Html.escapeHtml(
              `Company Number: ${props.feedback.type === 'companyFound' && props.feedback.company.number}`
            )}
          </p>
          <p>{Html.escapeHtml(`Email Address: ${props.feedback.type === 'companyFound' && props.email}`)}</p>
          <p>After clicking submit, a connection invitation will be sent to their email and postal address.</p>
        </div>
      </this.newConnectionForm>
    )
  }

  private newInviteSuccess = (props: { feedback: FormFeedback }): JSX.Element => {
    return (
      <this.newConnectionForm
        submitRoute="new/create-invitation"
        feedback={props.feedback}
        progressStep={3}
        progressStepCount={3}
        actions={[{ type: 'link', text: 'Back To Home', href: '/connection' }]}
      >
        <div id="new-connection-confirmation-text">
          <p>
            Your connection invitation has been sent. Please wait for their verification. As the post may take 2-3 days
            to arrive, please wait for their verification and keep updated by viewing the verification status.
          </p>
        </div>
      </this.newConnectionForm>
    )
  }
}
