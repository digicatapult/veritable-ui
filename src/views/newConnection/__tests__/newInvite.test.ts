import { expect } from 'chai'
import { describe, it } from 'mocha'

import { NewInviteTemplates } from '../newInvite.js'
import { successResponse, testErrorTargetBox, testMessageTargetBox, testSuccessTargetBox } from './fixtures.js'

describe('NewInviteTemplates', () => {
  describe('show form', () => {
    it('should render form with a error message and invalid response', async () => {
      const templates = new NewInviteTemplates()
      const rendered = await templates.newInviteForm({ feedback: testMessageTargetBox, formStage: 'form' })
      expect(rendered).to.matchSnapshot()
    })

    it('should render form with a valid response', async () => {
      const templates = new NewInviteTemplates()
      const rendered = await templates.newInviteForm({ feedback: testSuccessTargetBox, formStage: 'form' })
      expect(rendered).to.matchSnapshot()
    })

    it('should render a confirmation page with given email and company number', async () => {
      const templates = new NewInviteTemplates()
      const rendered = await templates.newInviteForm({
        feedback: testSuccessTargetBox,
        formStage: 'confirmation',
        email: '123@123.com',
        companyNumber: successResponse.company_number,
      })
      expect(rendered).to.matchSnapshot()
    })

    it('should render a success response page with a single button to return to home', async () => {
      const templates = new NewInviteTemplates()
      const rendered = await templates.newInviteForm({ feedback: testSuccessTargetBox, formStage: 'success' })
      expect(rendered).to.matchSnapshot()
    })

    it('should a web page with the a form in an empty state', async () => {
      const templates = new NewInviteTemplates()
      const rendered = await templates.newInviteFormPage(testMessageTargetBox)
      expect(rendered).to.matchSnapshot()
    })

    it('should a web page with the a form in an error state', async () => {
      const templates = new NewInviteTemplates()
      const rendered = await templates.newInviteFormPage(testErrorTargetBox)
      expect(rendered).to.matchSnapshot()
    })
  })
})
