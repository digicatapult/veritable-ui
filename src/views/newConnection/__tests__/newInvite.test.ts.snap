// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewInviteTemplates show form should a web page with the a form in an empty state 1`] = `"<!DOCTYPE html><html lang=\\"en\\"><head><script src=\\"/lib/htmx.org/htmx.min.js\\"></script><script src=\\"/lib/htmx-ext-json-enc/json-enc.js\\"></script><script src=\\"/public/scripts/auth-redirect.js\\"></script><link rel=\\"icon\\" type=\\"image/ico\\" href=\\"/public/images/favicon-dark.ico\\" media=\\"(prefers-color-scheme: dark)\\"/><link rel=\\"icon\\" type=\\"image/ico\\" href=\\"/public/images/favicon.ico\\" media=\\"(prefers-color-scheme: light), (prefers-color-scheme: no-preference)\\"/><link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"/public/styles/main.css\\"/><link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"/public/styles/connection.css\\"/><title>Veritable - New Connection</title></head><body hx-ext=\\"json-enc\\"><nav id=\\"side-bar\\"><a id=\\"veritable-logo\\" href=\\"/about\\"></a><div class=\\"profile-icon\\" style=\\"background-color:#38b6ff;\\">A</div><a title=\\"categories\\" href=\\"/\\" style=\\"--background-image:url('/public/images/category.svg');\\"></a><a title=\\"connections\\" href=\\"/connection\\" data-active style=\\"--background-image:url('/public/images/connection.svg');\\"></a><a title=\\"queries\\" href=\\"/queries\\" style=\\"--background-image:url('/public/images/query.svg');\\"></a><a title=\\"credentials\\" href=\\"/credentials\\" style=\\"--background-image:url('/public/images/credential.svg');\\"></a><a title=\\"settings\\" href=\\"/settings\\" style=\\"--background-image:url('/public/images/setting.svg');\\"></a></nav><main><div id=\\"content-header\\"><h1>New Connection</h1><div id=\\"content-header-nav\\"><a title=\\"home\\" class=\\"nav icon\\" href=\\"/\\"></a><a title=\\"Connections\\" href=\\"/connection\\">Connections</a><a title=\\"Invite New Connection\\" href=\\"/connection/new\\">Invite New Connection</a></div></div><div id=\\"content-main\\"><div class=\\"connections header\\"><span>Invite New Connection</span></div><div class=\\"card-body\\"><form id=\\"new-connection-form\\" hx-post=\\"/connection/new/create-invitation\\" hx-swap=\\"outerHTML\\" hx-select=\\"#new-connection-form\\"><div id=\\"new-connection-progress\\" style=\\"--progress-percent: 0.3333333333333333\\"><div id=\\"new-connection-progress-text\\"><span>Step 1</span><span> of 3</span></div><div id=\\"new-connection-progress-bar\\"></div></div><select id=\\"new-invite-country-select\\" name=\\"registryCountryCode\\" required hx-get=\\"/connection/new\\" hx-trigger=\\"change\\" hx-target=\\"#new-invite-company-number-input\\" hx-select=\\"#new-invite-company-number-input\\" hx-swap=\\"outerHTML\\" hx-include=\\"this\\"><option value=\\"GB\\">United Kingdom</option><option value=\\"US\\">US</option></select><input id=\\"new-invite-company-number-input\\" name=\\"companyNumber\\" placeholder=\\"Company House Number\\" required hx-get=\\"/connection/new/verify-company\\" hx-trigger=\\"keyup changed delay:200ms, change, load\\" hx-target=\\"#new-connection-feedback\\" hx-select=\\"#new-connection-feedback\\" hx-swap=\\"outerHTML\\" hx-include=\\"#new-invite-country-select\\" pattern=\\"^(((AC|CE|CS|FC|FE|GE|GS|IC|LP|NC|NF|NI|NL|NO|NP|OC|OE|PC|R0|RC|SA|SC|SE|SF|SG|SI|SL|SO|SR|SZ|ZC|\\\\d{2})\\\\d{6})|((IP|SP|RS)[A-Z\\\\d]{6})|(SL\\\\d{5}[\\\\dA]))$\\" minlength=\\"6\\" maxlength=\\"10\\" oninput=\\"this.reportValidity()\\" type=\\"text\\"/><input required id=\\"new-invite-email-input\\" placeholder=\\"Connection's Email Address\\" name=\\"email\\" type=\\"email\\"/><div id=\\"new-connection-feedback\\" class=\\"accented-container feedback-neutral\\">This is a message</div><div id=\\"new-connection-actions\\"><a class=\\"button \\" href=\\"/connection\\" data-variant=\\"outlined\\"><span>Cancel</span></a><button class=\\"button  \\" data-variant=\\"filled\\" type=\\"submit\\" name=\\"action\\" value=\\"continue\\" text=\\"Continue\\" style=\\"filled\\"><span>Continue</span></button></div></form></div></div></main></body></html>"`;

exports[`NewInviteTemplates show form should a web page with the a form in an error state 1`] = `"<!DOCTYPE html><html lang=\\"en\\"><head><script src=\\"/lib/htmx.org/htmx.min.js\\"></script><script src=\\"/lib/htmx-ext-json-enc/json-enc.js\\"></script><script src=\\"/public/scripts/auth-redirect.js\\"></script><link rel=\\"icon\\" type=\\"image/ico\\" href=\\"/public/images/favicon-dark.ico\\" media=\\"(prefers-color-scheme: dark)\\"/><link rel=\\"icon\\" type=\\"image/ico\\" href=\\"/public/images/favicon.ico\\" media=\\"(prefers-color-scheme: light), (prefers-color-scheme: no-preference)\\"/><link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"/public/styles/main.css\\"/><link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"/public/styles/connection.css\\"/><title>Veritable - New Connection</title></head><body hx-ext=\\"json-enc\\"><nav id=\\"side-bar\\"><a id=\\"veritable-logo\\" href=\\"/about\\"></a><div class=\\"profile-icon\\" style=\\"background-color:#38b6ff;\\">A</div><a title=\\"categories\\" href=\\"/\\" style=\\"--background-image:url('/public/images/category.svg');\\"></a><a title=\\"connections\\" href=\\"/connection\\" data-active style=\\"--background-image:url('/public/images/connection.svg');\\"></a><a title=\\"queries\\" href=\\"/queries\\" style=\\"--background-image:url('/public/images/query.svg');\\"></a><a title=\\"credentials\\" href=\\"/credentials\\" style=\\"--background-image:url('/public/images/credential.svg');\\"></a><a title=\\"settings\\" href=\\"/settings\\" style=\\"--background-image:url('/public/images/setting.svg');\\"></a></nav><main><div id=\\"content-header\\"><h1>New Connection</h1><div id=\\"content-header-nav\\"><a title=\\"home\\" class=\\"nav icon\\" href=\\"/\\"></a><a title=\\"Connections\\" href=\\"/connection\\">Connections</a><a title=\\"Invite New Connection\\" href=\\"/connection/new\\">Invite New Connection</a></div></div><div id=\\"content-main\\"><div class=\\"connections header\\"><span>Invite New Connection</span></div><div class=\\"card-body\\"><form id=\\"new-connection-form\\" hx-post=\\"/connection/new/create-invitation\\" hx-swap=\\"outerHTML\\" hx-select=\\"#new-connection-form\\"><div id=\\"new-connection-progress\\" style=\\"--progress-percent: 0.3333333333333333\\"><div id=\\"new-connection-progress-text\\"><span>Step 1</span><span> of 3</span></div><div id=\\"new-connection-progress-bar\\"></div></div><select id=\\"new-invite-country-select\\" name=\\"registryCountryCode\\" required hx-get=\\"/connection/new\\" hx-trigger=\\"change\\" hx-target=\\"#new-invite-company-number-input\\" hx-select=\\"#new-invite-company-number-input\\" hx-swap=\\"outerHTML\\" hx-include=\\"this\\"><option value=\\"GB\\">United Kingdom</option><option value=\\"US\\">US</option></select><input id=\\"new-invite-company-number-input\\" name=\\"companyNumber\\" placeholder=\\"Company House Number\\" required hx-get=\\"/connection/new/verify-company\\" hx-trigger=\\"keyup changed delay:200ms, change, load\\" hx-target=\\"#new-connection-feedback\\" hx-select=\\"#new-connection-feedback\\" hx-swap=\\"outerHTML\\" hx-include=\\"#new-invite-country-select\\" pattern=\\"^(((AC|CE|CS|FC|FE|GE|GS|IC|LP|NC|NF|NI|NL|NO|NP|OC|OE|PC|R0|RC|SA|SC|SE|SF|SG|SI|SL|SO|SR|SZ|ZC|\\\\d{2})\\\\d{6})|((IP|SP|RS)[A-Z\\\\d]{6})|(SL\\\\d{5}[\\\\dA]))$\\" minlength=\\"6\\" maxlength=\\"10\\" oninput=\\"this.reportValidity()\\" type=\\"text\\"/><input required id=\\"new-invite-email-input\\" placeholder=\\"Connection's Email Address\\" name=\\"email\\" type=\\"email\\"/><div id=\\"new-connection-feedback\\" class=\\"accented-container feedback-negative\\">This is a test error message</div><div id=\\"new-connection-actions\\"><a class=\\"button \\" href=\\"/connection\\" data-variant=\\"outlined\\"><span>Cancel</span></a><button class=\\"button  \\" data-variant=\\"filled\\" type=\\"submit\\" name=\\"action\\" value=\\"continue\\" text=\\"Continue\\" style=\\"filled\\"><span>Continue</span></button></div></form></div></div></main></body></html>"`;

exports[`NewInviteTemplates show form should render a confirmation page with given email and company number 1`] = `"<form id=\\"new-connection-form\\" hx-post=\\"/connection/new/create-invitation\\" hx-swap=\\"outerHTML\\" hx-select=\\"#new-connection-form\\"><div id=\\"new-connection-progress\\" style=\\"--progress-percent: 0.6666666666666666\\"><div id=\\"new-connection-progress-text\\"><span>Step 2</span><span> of 3</span></div><div id=\\"new-connection-progress-bar\\"></div></div><input id=\\"new-invite-company-number-input\\" name=\\"companyNumber\\" value=\\"07964699\\" type=\\"hidden\\"/><input id=\\"new-invite-email-input\\" name=\\"email\\" value=\\"123@123.com\\" type=\\"hidden\\"/><input id=\\"new-invite-country-code-input\\" name=\\"registryCountryCode\\" value=\\"GB\\" type=\\"hidden\\"/><div id=\\"new-connection-confirmation-text\\"><p>Please confirm the details of the connection before sending</p><p> GB</p><p>Company House Number: 07964699</p><p>Email Address: 123@123.com</p><p>After clicking submit, a connection invitation will be sent to their email and postal address.</p></div><div id=\\"new-connection-feedback\\" class=\\"accented-container feedback-positive\\"><div><p><span>Registered Office Address</span><span class=\\"address-line\\">Level 9,&nbsp</span><span class=\\"address-line\\"> 101 Euston Road,&nbsp</span><span class=\\"address-line\\"> London,&nbsp</span><span class=\\"address-line\\"> NW1 2RA</span></p><p><span>Company Status</span>active</p></div><div><img src=\\"/public/images/check.svg\\" alt=\\"Description of the image\\"/></div></div><div id=\\"new-connection-actions\\"><button class=\\"button  \\" data-variant=\\"outlined\\" type=\\"submit\\" name=\\"action\\" value=\\"back\\" text=\\"Back\\" style=\\"outlined\\"><span>Back</span></button><button class=\\"button  \\" data-variant=\\"filled\\" type=\\"submit\\" name=\\"action\\" value=\\"submit\\" text=\\"Submit\\" style=\\"filled\\"><span>Submit</span></button></div></form>"`;

exports[`NewInviteTemplates show form should render a success response page with a single button to return to home 1`] = `"<form id=\\"new-connection-form\\" hx-post=\\"/connection/new/create-invitation\\" hx-swap=\\"outerHTML\\" hx-select=\\"#new-connection-form\\"><div id=\\"new-connection-progress\\" style=\\"--progress-percent: 1\\"><div id=\\"new-connection-progress-text\\"><span>Step 3</span><span> of 3</span></div><div id=\\"new-connection-progress-bar\\"></div></div><div id=\\"new-connection-confirmation-text\\"><p>Your connection invitation has been sent. Please wait for their verification. As the post may take 2-3 days to arrive, please wait for their verification and keep updated by viewing the verification status.</p></div><div id=\\"new-connection-feedback\\" class=\\"accented-container feedback-positive\\"><div><p><span>Registered Office Address</span><span class=\\"address-line\\">Level 9,&nbsp</span><span class=\\"address-line\\"> 101 Euston Road,&nbsp</span><span class=\\"address-line\\"> London,&nbsp</span><span class=\\"address-line\\"> NW1 2RA</span></p><p><span>Company Status</span>active</p></div><div><img src=\\"/public/images/check.svg\\" alt=\\"Description of the image\\"/></div></div><div id=\\"new-connection-actions\\"><a class=\\"button \\" href=\\"/connection\\" data-variant=\\"filled\\"><span>Back To Home</span></a></div></form>"`;

exports[`NewInviteTemplates show form should render form with a error message and invalid response 1`] = `"<form id=\\"new-connection-form\\" hx-post=\\"/connection/new/create-invitation\\" hx-swap=\\"outerHTML\\" hx-select=\\"#new-connection-form\\"><div id=\\"new-connection-progress\\" style=\\"--progress-percent: 0.3333333333333333\\"><div id=\\"new-connection-progress-text\\"><span>Step 1</span><span> of 3</span></div><div id=\\"new-connection-progress-bar\\"></div></div><select id=\\"new-invite-country-select\\" name=\\"registryCountryCode\\" required hx-get=\\"/connection/new\\" hx-trigger=\\"change\\" hx-target=\\"#new-invite-company-number-input\\" hx-select=\\"#new-invite-company-number-input\\" hx-swap=\\"outerHTML\\" hx-include=\\"this\\"><option value=\\"GB\\">United Kingdom</option><option value=\\"US\\">US</option></select><input id=\\"new-invite-company-number-input\\" name=\\"companyNumber\\" placeholder=\\"Company House Number\\" required hx-get=\\"/connection/new/verify-company\\" hx-trigger=\\"keyup changed delay:200ms, change, load\\" hx-target=\\"#new-connection-feedback\\" hx-select=\\"#new-connection-feedback\\" hx-swap=\\"outerHTML\\" hx-include=\\"#new-invite-country-select\\" pattern=\\"^(((AC|CE|CS|FC|FE|GE|GS|IC|LP|NC|NF|NI|NL|NO|NP|OC|OE|PC|R0|RC|SA|SC|SE|SF|SG|SI|SL|SO|SR|SZ|ZC|\\\\d{2})\\\\d{6})|((IP|SP|RS)[A-Z\\\\d]{6})|(SL\\\\d{5}[\\\\dA]))$\\" minlength=\\"6\\" maxlength=\\"10\\" oninput=\\"this.reportValidity()\\" type=\\"text\\"/><input required id=\\"new-invite-email-input\\" placeholder=\\"Connection's Email Address\\" name=\\"email\\" type=\\"email\\"/><div id=\\"new-connection-feedback\\" class=\\"accented-container feedback-neutral\\">This is a message</div><div id=\\"new-connection-actions\\"><a class=\\"button \\" href=\\"/connection\\" data-variant=\\"outlined\\"><span>Cancel</span></a><button class=\\"button  \\" data-variant=\\"filled\\" type=\\"submit\\" name=\\"action\\" value=\\"continue\\" text=\\"Continue\\" style=\\"filled\\"><span>Continue</span></button></div></form>"`;

exports[`NewInviteTemplates show form should render form with a valid response 1`] = `"<form id=\\"new-connection-form\\" hx-post=\\"/connection/new/create-invitation\\" hx-swap=\\"outerHTML\\" hx-select=\\"#new-connection-form\\"><div id=\\"new-connection-progress\\" style=\\"--progress-percent: 0.3333333333333333\\"><div id=\\"new-connection-progress-text\\"><span>Step 1</span><span> of 3</span></div><div id=\\"new-connection-progress-bar\\"></div></div><select id=\\"new-invite-country-select\\" name=\\"registryCountryCode\\" required hx-get=\\"/connection/new\\" hx-trigger=\\"change\\" hx-target=\\"#new-invite-company-number-input\\" hx-select=\\"#new-invite-company-number-input\\" hx-swap=\\"outerHTML\\" hx-include=\\"this\\"><option value=\\"GB\\">United Kingdom</option><option value=\\"US\\">US</option></select><input id=\\"new-invite-company-number-input\\" name=\\"companyNumber\\" placeholder=\\"Company House Number\\" required hx-get=\\"/connection/new/verify-company\\" hx-trigger=\\"keyup changed delay:200ms, change, load\\" hx-target=\\"#new-connection-feedback\\" hx-select=\\"#new-connection-feedback\\" hx-swap=\\"outerHTML\\" hx-include=\\"#new-invite-country-select\\" pattern=\\"^(((AC|CE|CS|FC|FE|GE|GS|IC|LP|NC|NF|NI|NL|NO|NP|OC|OE|PC|R0|RC|SA|SC|SE|SF|SG|SI|SL|SO|SR|SZ|ZC|\\\\d{2})\\\\d{6})|((IP|SP|RS)[A-Z\\\\d]{6})|(SL\\\\d{5}[\\\\dA]))$\\" minlength=\\"6\\" maxlength=\\"10\\" oninput=\\"this.reportValidity()\\" type=\\"text\\"/><input required id=\\"new-invite-email-input\\" placeholder=\\"Connection's Email Address\\" name=\\"email\\" type=\\"email\\"/><div id=\\"new-connection-feedback\\" class=\\"accented-container feedback-positive\\"><div><p><span>Registered Office Address</span><span class=\\"address-line\\">Level 9,&nbsp</span><span class=\\"address-line\\"> 101 Euston Road,&nbsp</span><span class=\\"address-line\\"> London,&nbsp</span><span class=\\"address-line\\"> NW1 2RA</span></p><p><span>Company Status</span>active</p></div><div><img src=\\"/public/images/check.svg\\" alt=\\"Description of the image\\"/></div></div><div id=\\"new-connection-actions\\"><a class=\\"button \\" href=\\"/connection\\" data-variant=\\"outlined\\"><span>Cancel</span></a><button class=\\"button  \\" data-variant=\\"filled\\" type=\\"submit\\" name=\\"action\\" value=\\"continue\\" text=\\"Continue\\" style=\\"filled\\"><span>Continue</span></button></div></form>"`;
